src/avl_tree.jl

```diff
@@ -0,0 +1,259 @@
+# it has unique keys
+# leftChild has keys which are less than the node
+# rightChild has keys which are greater than the node
+# height stores the height of the subtree.
+mutable struct AVLTreeNode{K}
+    height::Int8
```

Comment:
Is Int8 big enough?
that means max height is 128.
Is it so self-balanced that that means 2^128ish items (which is more than enough)?

Reply:
Well, it is the smallest `Integer` type right ?

Reply:
right.